test/hiccup2/optimizations_test.clj

```diff
@@ -0,0 +1,90 @@
+(ns hiccup2.optimizations-test
+  (:require [clojure.pprint :as pp]
+            [clojure.test :refer :all]
+            [clojure.walk :as walk]
+            [hiccup2.core :as h]))
+
+(defn- count-forms [data]
+  (let [*counter (atom 0)]
+    (walk/postwalk (fn [x]
+                     (when (seq? x)
+                       (swap! *counter inc))
+                     x)
+                   data)
+    @*counter))
```

Comment:
What about:

```clojure
(defn- count-forms [data]
  (count (filter seq? (tree-seq coll? seq data))))
```

I believe it'll generate the same result, just a little more functionally.